File: C#/Graph - BFS/1926. Nearest Exit from Entrance in Maze.cs
/* 1926. Nearest Exit from Entrance in Maze 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/nearest-exit-from-entrance-in-maze/submissions/1382353476/ */

public class Solution {
    public int NearestExit(char[][] maze, int[] entrance) {
        
        var directions = new int[][]
        {
            new int[]{0,1},
            new int[]{1,0},
            new int[]{0,-1},
            new int[]{-1,0}
        };

        var queue = new Queue<int[]>();
        queue.Enqueue(new int[]{entrance[0], entrance[1], 0});
        maze[entrance[0]][entrance[1]] = '+';

        while(queue.Count > 0)
        {
            var current = queue.Dequeue();
            int row = current[0], col = current[1], steps = current[2];

            foreach(var direction in directions)
            {
                var newRow = direction[0] + row;
                var newCol = direction[1] + col;

                if(newRow >= 0 && newRow < maze.Length  && newCol >= 0 && newCol < maze[0].Length && maze[newRow][newCol] == '.' )
                {
                    if(newRow == 0 || newRow == maze.Length - 1  || newCol == 0 || newCol == maze[0].Length - 1 )
                    {
                        return steps + 1;
                    }

                    maze[newRow][newCol] = '+';
                    queue.Enqueue(new int[]{newRow, newCol, steps+1});
                }
            }
        }
       
       return -1;
    }
}

File: Python/Graph - BFS/1926. Nearest Exit from Entrance in Maze.py
""" 1926. Nearest Exit from Entrance in Maze 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/nearest-exit-from-entrance-in-maze/submissions/1382352939/ """

class Solution(object):
    def nearestExit(self, maze, entrance):
        m=len(maze)
        n=len(maze[0])
        directions=[(-1,0),(1,0),(0,-1),(0,1)]
        queue=deque()
        queue.append([entrance[0],entrance[1],0])
        maze[entrance[0]][entrance[1]] = '+'
        while queue:
            elex,eley,dist=queue.popleft()
            for x,y in directions:

                x1=elex+x
                y1=eley+y
                if 0<=x1<=m-1 and 0<=y1<=n-1 and maze[x1][y1]=='.':
                    if x1==0 or x1==m-1 or y1==0 or y1==n-1:
                        return dist+1

                    else:
                        maze[x1][y1] = '+'
                        queue.append([x1,y1,dist+1])

        return -1